frontend/src/routes/(authenticated)/admin/+page.ts

```diff
@@ -3,43 +3,70 @@ import { getClient, graphql } from '$lib/gql';
 import type { PageLoadEvent } from './$types';
 import { isAdmin, type LexAuthUser } from '$lib/user';
 import { redirect } from '@sveltejs/kit';
-import { getBoolSearchParam } from '$lib/util/query-params';
-import type { $OpResult, ChangeUserAccountByAdminInput, ChangeUserAccountByAdminMutation, ProjectType } from '$lib/gql/types';
-import type {LoadAdminDashboardQuery} from '$lib/gql/types';
+import { getBoolSearchParam, getSearchParam } from '$lib/util/query-params';
+import type { $OpResult, ChangeUserAccountByAdminInput, ChangeUserAccountByAdminMutation, ProjectFilterInput, ProjectType } from '$lib/gql/types';
+import type { LoadAdminDashboardProjectsQuery, LoadAdminDashboardUsersQuery } from '$lib/gql/types';
 
-export type AdminSearchParams = {
+export const _FILTER_PAGE_SIZE = 100;
+
+export type AdminProjectSearchParams = {
   showDeletedProjects: boolean,
   projectType: ProjectType | undefined,
   userEmail: string | undefined,
   projectSearch: string,
 };
 
-export type Project = LoadAdminDashboardQuery['projects'][number];
-export type User = LoadAdminDashboardQuery['users']['items'][number];
+export type AdminUserSearchParams = {
+  userSearch: string,
+  userEmail: string | undefined,
+}
+
+export type AdminSearchParams = AdminProjectSearchParams & AdminUserSearchParams;
+
+export type Project = LoadAdminDashboardProjectsQuery['projects'][number];
+export type User = NonNullable<NonNullable<LoadAdminDashboardUsersQuery['users']>['items']>[number];
 
 export async function load(event: PageLoadEvent) {
   const parentData = await event.parent();
   requireAdmin(parentData.user);
 
   const withDeletedProjects = getBoolSearchParam<AdminSearchParams>('showDeletedProjects', event.url.searchParams);
+  const userSearch = getSearchParam<AdminSearchParams>('userSearch', event.url.searchParams) ?? '';
+  const userEmail = getSearchParam<AdminSearchParams>('userEmail', event.url.searchParams);
 
   const client = getClient();
+
+  const projectFilter = {
+    ...(userEmail ? { users: { some: { user: { email: { icontains: userEmail } } } } } : {})
+  } as ProjectFilterInput;
```

Comment:
I changed this:
~`avoidOptionals: true,`~
`avoidOptionals: false,`
to make that possible. Otherwise it expected `null` to be explicitly passed for tons of unnecessary values.